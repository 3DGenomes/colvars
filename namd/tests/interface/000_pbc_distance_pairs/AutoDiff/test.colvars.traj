# step         dmat                                                                                              
           0   (  3.76775588938397e+00 ,  1.27670424199830e+01 ,  1.56085605312159e+01 ,  3.81804080950135e+00 ) 
           1   (  3.77332907564893e+00 ,  1.27560493859334e+01 ,  1.56085153724134e+01 ,  3.82452982766156e+00 ) 
           2   (  3.78019975969672e+00 ,  1.27468636648733e+01 ,  1.56111277075737e+01 ,  3.83175022864051e+00 ) 
           3   (  3.78819771933008e+00 ,  1.27397695334328e+01 ,  1.56161953754710e+01 ,  3.83932346865777e+00 ) 
           4   (  3.79708817984111e+00 ,  1.27349749143542e+01 ,  1.56234011760942e+01 ,  3.84689315912478e+00 ) 
           5   (  3.80658512542060e+00 ,  1.27326068589452e+01 ,  1.56323288068563e+01 ,  3.85414817220345e+00 ) 
           6   (  3.81636795219360e+00 ,  1.27327048218882e+01 ,  1.56424874429791e+01 ,  3.86084223100772e+00 ) 
           7   (  3.82610015667133e+00 ,  1.27352116080287e+01 ,  1.56533460803375e+01 ,  3.86680701490909e+00 ) 
           8   (  3.83544892725488e+00 ,  1.27399641116498e+01 ,  1.56643739772086e+01 ,  3.87195803203618e+00 ) 
           9   (  3.84410469585822e+00 ,  1.27466879412248e+01 ,  1.56750870862394e+01 ,  3.87629118404174e+00 ) 
          10   (  3.85179983313886e+00 ,  1.27550008330810e+01 ,  1.56850887367242e+01 ,  3.87987153704871e+00 ) 
          11   (  3.85832570912291e+00 ,  1.27644287239806e+01 ,  1.56940949807935e+01 ,  3.88281727391239e+00 ) 
          12   (  3.86354718260077e+00 ,  1.27744354609444e+01 ,  1.57019498365685e+01 ,  3.88527980376443e+00 ) 
          13   (  3.86741303928011e+00 ,  1.27844640769184e+01 ,  1.57086306268560e+01 ,  3.88742217552241e+00 ) 
          14   (  3.86995956288113e+00 ,  1.27939866840406e+01 ,  1.57142438618352e+01 ,  3.88939723861039e+00 ) 
          15   (  3.87130879799583e+00 ,  1.28025512568263e+01 ,  1.57190119924486e+01 ,  3.89132602596511e+00 ) 
          16   (  3.87165970709109e+00 ,  1.28098164444715e+01 ,  1.57232514449873e+01 ,  3.89327812532905e+00 ) 
          17   (  3.87127160503671e+00 ,  1.28155724745827e+01 ,  1.57273429570682e+01 ,  3.89525532071078e+00 ) 
          18   (  3.87044062317139e+00 ,  1.28197478763871e+01 ,  1.57316962894983e+01 ,  3.89718128877117e+00 ) 
          19   (  3.86947121952079e+00 ,  1.28224035550395e+01 ,  1.57367127978342e+01 ,  3.89890092197056e+00 ) 
          20   (  3.86864602891868e+00 ,  1.28237173481025e+01 ,  1.57427501720568e+01 ,  3.90019195054618e+00 ) 
