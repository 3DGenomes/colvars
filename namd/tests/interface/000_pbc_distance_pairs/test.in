colvarsTrajFrequency    1

colvar {
    name dmat

    width 0.2

    distancePairs {
        group1 {
            indexGroup Protein_C-alpha_1
        }
        group2 {
            indexGroup Protein_C-alpha_2
        }
    }
} 


harmonic {
    colvars        dmat
    forceConstant  0.01
    centers        (5.0, 120.0, 150.0, 5.0)
    targetCenters  (5.0, 5.0, 5.0, 5.0)
    targetNumSteps 1000
}
