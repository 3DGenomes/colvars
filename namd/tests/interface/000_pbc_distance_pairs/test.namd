# -*- tcl -*-

set mol_name "15ala2x"
set pbc "no"
set coor_pdb_file "../Common/${mol_name}.pdb"
set coor_bin_file "../Common/${mol_name}.init.coor"
set vel_bin_file "../Common/${mol_name}.init.vel"
set xsc_file "../Common/${mol_name}.init.xsc"
set par_files [list "../Common/par_all36_prot.prm"]

proc num_atoms {} {
    if { ${mol_name} == "15ala2x" } {
        return 324
    }
    if { ${mol_name} == "15leu2x" } {
        return 594
    }
    return -1
}

if { [llength [info commands print]] > 0 } {

    # NAMD
    source ../Common/common.namd

    DCDfile test.dcd
    DCDfreq 1
    DCDUnitCell yes

    colvars on

} else {

    # VMD

    mol new ../Common/${mol_name}.psf type psf
    mol addfile ${coor_bin_file} type namdbin waitfor all
    mol addfile test.dcd type dcd waitfor all
    cv molid top
}


cv config "indexFile ../Common/${mol_name}.ndx"
cv configfile test.in


if { [llength [info commands print]] > 0 } {
    # NAMD
    run ${numsteps}
} else {
    # VMD
    set out [open "test.colvars.traj" "w"]
    puts -nonewline ${out} [cv printframelabels]
    for { set fr 0 } { ${fr} < [molinfo top get numframes] } { incr fr } {
        cv frame ${fr}
        cv update
        puts -nonewline ${out} [cv printframe]
    }
    close ${out}
    quit
}

