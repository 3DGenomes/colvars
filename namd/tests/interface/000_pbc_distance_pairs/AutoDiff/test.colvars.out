colvars: ----------------------------------------------------------------------
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:  http://dx.doi.org/10.1080/00268976.2013.813594
colvars: in any publication based on this calculation.
colvars: SMP parallelism is available.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration:
colvars: # indexFile = "../Common/15ala2x.ndx"
colvars: The following index groups were read from the index file "../Common/15ala2x.ndx":
colvars:   Protein_HA (162 atoms).
colvars:   Protein_noH_HA (80 atoms).
colvars:   Protein_Backbone_HA (60 atoms).
colvars:   Protein_C-alpha_HA (15 atoms).
colvars:   RMSD_atoms_HA (15 atoms).
colvars:   Protein_HB (162 atoms).
colvars:   Protein_noH_HB (80 atoms).
colvars:   Protein_Backbone_HB (60 atoms).
colvars:   Protein_C-alpha_HB (15 atoms).
colvars:   RMSD_atoms_HB (15 atoms).
colvars:   Protein_C-alpha_1_2 (4 atoms).
colvars:   Protein_C-alpha_9_10 (4 atoms).
colvars:   Protein_C-alpha_1 (2 atoms).
colvars:   group1 (8 atoms).
colvars:   Protein_C-alpha_2 (2 atoms).
colvars:   group2 (8 atoms).
colvars:   Protein_C-alpha_3 (2 atoms).
colvars:   group3 (8 atoms).
colvars:   Protein_C-alpha_4 (2 atoms).
colvars:   group4 (8 atoms).
colvars:   Protein_C-alpha_5 (2 atoms).
colvars:   group5 (8 atoms).
colvars:   Protein_C-alpha_6 (2 atoms).
colvars:   group6 (8 atoms).
colvars:   Protein_C-alpha_7 (2 atoms).
colvars:   group7 (8 atoms).
colvars:   Protein_C-alpha_8 (2 atoms).
colvars:   group8 (8 atoms).
colvars:   Protein_C-alpha_9 (2 atoms).
colvars:   group9 (8 atoms).
colvars:   Protein_C-alpha_10 (2 atoms).
colvars:   group10 (8 atoms).
colvars:   heavy_atoms (160 atoms).
colvars: # smp = on [default]
colvars: # analysis = off [default]
colvars: # colvarsTrajFrequency = 100 [default]
colvars: # colvarsRestartFrequency = 100 [default]
colvars: # colvarsTrajAppend = off [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = on [default]
colvars: Warning: no collective variables defined.
colvars: Collective variables initialized, 0 in total.
colvars: No collective variables biases were defined.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "test.in":
colvars: # indexFile = "" [default]
colvars: # smp = on [default]
colvars: # analysis = off [default]
colvars: # colvarsTrajFrequency = 1
colvars: # colvarsRestartFrequency = 100 [default]
colvars: # colvarsTrajAppend = off [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = on [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "dmat"
colvars:   Initializing a new "distancePairs" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # scalable = on [default]
colvars:     # forceNoPBC = off [default]
colvars:       Initializing atom group "group1".
colvars:       # indexGroup = "Protein_C-alpha_1"
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "group1" defined, 2 atoms initialized: total mass = 24.022, total charge = 0.14.
colvars:       Initializing atom group "group2".
colvars:       # indexGroup = "Protein_C-alpha_2"
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "group2" defined, 2 atoms initialized: total mass = 24.022, total charge = 0.14.
colvars:   All components initialized.
colvars:   # width = 0.2
colvars:   # expandBoundaries = off [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonic" instance.
colvars:   # name = "harmonic1" [default]
colvars:   # colvars = { dmat }
colvars:   # outputEnergy = off [default]
colvars:   # forceConstant = 0.01
colvars:   # centers = { ( 5 , 120 , 150 , 5 ) }
colvars:   # targetCenters = { ( 5 , 5 , 5 , 5 ) }
colvars:   # targetForceConstant = 0 [default]
colvars:   # targetNumSteps = 1000
colvars:   # targetNumStages = 0 [default]
colvars:   # targetForceExponent = 1 [default]
colvars:   # outputCenters = off [default]
colvars:   # outputAccumulatedWork = off [default]
colvars:   The force constant for colvar "dmat" will be rescaled to 0.25 according to the specified width.
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Updating NAMD interface:
colvars: updating atomic data (4 atoms).
colvars: updating group data (0 scalable groups, 0 atoms in total).
colvars: Updating NAMD interface:
colvars: updating atomic data (4 atoms).
colvars: updating group data (0 scalable groups, 0 atoms in total).
colvars: Re-initialized atom group dmat:0/0. 2 atoms: total mass = 24.022.
colvars: Re-initialized atom group dmat:0/1. 2 atoms: total mass = 24.022.
colvars: The restart output state file will be "test.tmp.colvars.state".
colvars: The final output state file will be "test.colvars.state".
colvars: Writing to colvar trajectory file "test.colvars.traj".
colvars: Saving collective variables state to "test.colvars.state".
